aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -268,66 +268,43 @@ public synchronized void initialize(
+                final ModelSchema modelSchema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
+
+                final StorageItemChange.Type writeType;
+                if (modelExists(item, QueryPredicates.all())) {
+                    // if data exists already, then UPDATE the row
+                    writeType = StorageItemChange.Type.UPDATE;
                 } else if (!QueryPredicates.all().equals(predicate)) {
                     // insert not permitted with a condition
-                    onError.accept(new DataStoreException(
+                    throw new DataStoreException(
                         "Conditional update must be performed against an already existing data. " +
                             "Insertion is not permitted while using a predicate.",
                         "Please save without specifying a predicate."
-                    ));
-                    return;
+                    );
                 } else {
-                    // insert model in SQLite
-                    type = StorageItemChange.Type.CREATE;
+                    // if data doesn't exist yet, then INSERT a new row
+                    writeType = StorageItemChange.Type.CREATE;
+                }
 
-                    sqlCommand = sqlCommandFactory.insertFor(modelSchema);
-                    if (!sqlCommand.hasCompiledSqlStatement()) {
-                        onError.accept(new DataStoreException(
-                            "No insert statement found for the Model: " + modelSchema.getName(),
-                            AmplifyException.TODO_RECOVERY_SUGGESTION
-                        ));
-                        return;
-                    }
-                    modelConflictStrategy = ModelConflictStrategy.THROW_EXCEPTION;
+                // Check if existing data meets the condition
+                if (StorageItemChange.Type.UPDATE.equals(writeType) && !modelExists(item, predicate)) {
+                    throw new DataStoreException(
+                        "Save failed because condition did not match existing model instance.",
+                        "The save will continue to fail until the model instance is updated."
+                    );
                 }
 
-                saveModel(item, modelSchema, sqlCommand, modelConflictStrategy);
-                final StorageItemChange<T> change = StorageItemChange.<T>builder()
-                    .changeId(item.getId())
-                    .item(item)
-                    .modelSchema(modelSchema)
-                    .type(type)
-                    .predicate(predicate)
-                    .initiator(initiator)
-                    .build();
+                // execute local save
+                writeData(item, writeType);
+
+                // publish successful save
+                StorageItemChange<T> change = StorageItemChange.<T>builder()
+                        .item(item)
```

Comment:
is `.changeId(item.getId())` missing here?

Reply:
the adapter used to set `changeId` to the actual ID of the model being mutated, but those are actually supposed to be two separate things to identify. The previous implementation didn't necessarily break anything, but this is how it should have been done in the first place.

@richardmcclellan can you confirm that the codebase doesn't make an assumption that `changeId` matches the model `id` anywhere else in the code?

Reply:
I don't believe this value is used at all internally.  However, it is included in the `DataStoreChangeEvent` which is sent back to customers, so it could break customers.  That said, since we are changing it to behave how it is supposed to behave according to the javadoc, I think it's okay to move forward with it.